azure-test/tests/azure_dns_zone/variables.tf

```
@@ -0,0 +1,68 @@
+
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20230313-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "azure_environment" {
+  type        = string
+  default     = "public"
+  description = "Azure environment used for the test."
+}
+
+variable "azure_subscription" {
+  type        = string
+  default     = "3510ae4d-530b-497d-8f30-53b9616fc6c1"
+  description = "Azure subscription used for the test."
+}
+
+provider "azurerm" {
+  # Cannot be passed as a variable
+  environment     = var.azure_environment
+  subscription_id = var.azure_subscription
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "azure:///subscriptions/${data.azurerm_client_config.current.subscription_id}"
+  }
+}
+
+resource "azurerm_resource_group" "named_test_resource" {
+  name     = var.resource_name
+  location = "East US"
+  tags = {
+    name = var.resource_name
+  }
+}
+
+resource "azurerm_dns_zone" "named_test_resource" {
+  name                = "${var.resource_name}.xyz"
```

```suggestion
  name                = "${var.resource_name}"
```
Do we need `.xyz` references?

---

Yes, it is required to be a valid DNS name with at least two parts.

---

FWIW, I took the `.xyz` from Azure documentation examples: https://learn.microsoft.com/en-us/azure/dns/dns-getstarted-portal